twilio/http/http_client.py

```diff
@@ -13,12 +13,26 @@ class TwilioHttpClient(HttpClient):
     """
     General purpose HTTP Client for interacting with the Twilio API
     """
-    def __init__(self, pool_connections=True, request_hooks=None):
+    def __init__(self, pool_connections=True, request_hooks=None, timeout=None):
+        """
+        Constructor for the TwilioHttpClient
+
+        :param bool pool_connections
+        :param request_hooks
+        :param int timeout: Timeout for the requests.
+                            Timeout should never be zero (0) or less.
+        """
         self.session = Session() if pool_connections else None
         self.last_request = None
         self.last_response = None
         self.request_hooks = request_hooks or hooks.default_hooks()
 
+        if timeout is None:
+            pass
+        elif timeout <= 0:
```

Comment:
`if timeout <= 0:`

Reply:
`None <= 0` evals to `true`.

Reply:
`if timeout is not None and timeout <= 0:`? Personally don't love this use of `pass`, but I could concede.